src/v/cluster/partition.cc

```diff
@@ -455,6 +533,33 @@ ss::future<> partition::remove_remote_persistent_state() {
     }
 }
 
+ss::future<> partition::stop_archiver() {
+    if (_archiver) {
+        return _archiver->stop().then([this] {
```

Comment:
Is there any risk of another `stop()` being scheduled between `stop()` and `archiver = nullptr`. Feels very unlikely though.